pandas/core/strings.py

```diff
@@ -1967,65 +1967,58 @@ def _get_series_list(self, others, ignore_index=False):
                    'only objects of type Series/Index/list-like/np.ndarray)')
 
         if isinstance(others, Series):
-            fu_wrn = not others.index.equals(idx)
+            warn = not others.index.equals(idx)
             los = [Series(others.values, index=idx)
-                   if ignore_index and fu_wrn else others]
-            return (los, fu_wrn)
+                   if ignore_index and warn else others]
+            return (los, warn)
         elif isinstance(others, Index):
-            fu_wrn = not others.equals(idx)
+            warn = not others.equals(idx)
             los = [Series(others.values,
                           index=(idx if ignore_index else others))]
-            return (los, fu_wrn)
+            return (los, warn)
         elif isinstance(others, DataFrame):
-            fu_wrn = not others.index.equals(idx)
-            if ignore_index and fu_wrn:
+            warn = not others.index.equals(idx)
+            if ignore_index and warn:
                 # without copy, this could change "others"
                 # that was passed to str.cat
                 others = others.copy()
                 others.index = idx
-            return ([others[x] for x in others], fu_wrn)
+            return ([others[x] for x in others], warn)
         elif isinstance(others, np.ndarray) and others.ndim == 2:
```

Comment:
@jreback :
> this is wrong
i don’t think we can align a ndarray at all like this
let’s can ndarray a that are > 1 dim

My response: 
The DF-constructor works as expected for a 2-dim ndarray, but I haven't checked if this is tested behaviour. (essentially, `df == DataFrame(df.values, columns=df.columns, index=df.index)`)

I would suggest not to can 2-dim ndarrays, because they are necessary to avoid alignment on the deprecation path for `join`:
> [...] To disable alignment (the behavior before v.0.23) and  silence this warning, use `.values` on any Series/Index/DataFrame in `others`. [...]

Reply:
can you add a test for this, basically see what happens when I have a non-default index with the Series (e.g. 2,3, 4) or something and it gets aligned with the 0,1,2 of the ndarray-converted-to-DataFrame. It will 'work' but is wrong.

Reply:
I added a general comment before all the cases for `others` because it seems this point wasn't clear:
```
# Generally speaking, all objects without an index inherit the index
# `idx` of the calling Series/Index - i.e. must have matching length.
# Objects with an index (i.d. Series/Index/DataFrame) keep their own
# index, *unless* ignore_index is set to True.
```

So, the case you mention does not happen, because an ndarray is always automatically align with the calling Series (of course, the lengths must match for this to work).

There are several tests with different indexes, both with `join is None` (so successful but raising a warning), as well as with the different `join`-types.

Reply:
i c, ok then